src/Controller/Cliente/ClienteFormController.php

```diff
@@ -20,31 +21,53 @@ public function __construct(ClienteService $clienteService)
     }
 
     #[Route("/clientes/adicionar", name: "RegistrarCliente", methods: ['POST'])]
-    public function registrarCliente(Request $request,ValidarCpfService $validarCpfService): Response
+    public function registrarCliente(Request $request,ValidarCpfService $validarCpfService, ClienteRepository $clienteRepository): Response
     {
         $token = $request->request->get("_csrf_token");
 
         if (!$this->isCsrfTokenValid("addCliente", $token)) {
             $this->addFlash("danger", "Token CRSF inválido!");
             return $this->redirectToRoute("clientes");
         }
+
+        if(!$validarCpfService->execute($request->request->get("cpf"))){
+            $this->addFlash("danger", "CPF inválido!");
+            return $this->redirectToRoute("adicionarCliente");
+        }
         
         $cliente = new Cliente();
 
-        if(!$validarCpfService->execute($request->request->get("cpf"))){
-            $this->addFlash("danger", "CPF invalido inválido!");
-            return $this->redirectToRoute("clientes");
+        $cpf = $request->request->get("cpf");
+        $cpf = str_replace(['.', '-'], '', $cpf);
+
+        if($clienteRepository->buscarClienteCPF($cpf)){
+            $this->addFlash("danger", "CPF já cadastrado!");
+            return $this->redirectToRoute("adicionarCliente");
         }
 
-        $cliente->setCpf((int) $request->request->get("cpf"));
-        $cliente->setNome((string) $request->request->get("nome"));
+        $cliente->setCpf((int) $cpf);
```

Comment:
sempre string, muda o tipo do parametro

```suggestion
        $cliente->setCpf($cpf);
```

Reply:
sem sets, vamos passar o nome e cpf por parametro no construtor, assim fica carantido que um objeto Cliente tem nome e cpf e representa ao abstrato